### PR TITLE
[Tutorials] Make converttonotebook smarter, and update RF tutorials.

### PR DESCRIPTION
**Pull request by hageboeck**
- Code in the df103 tutorial notebooks [here](https://nbviewer.jupyter.org/url/root.cern/doc/master/notebooks/df103_NanoAODHiggsAnalysis.C.nbconvert.ipynb) doesn't compile, since `converttonotebook.py` only recognises functions with a few hard-coded return types. Here, this limitation is lifted (or at least a broader range of functions is recognised ...).
- Using the new functionality, some RooFit tutorials could be updated.
- And a new tutorial was added to clarify [ROOT-8440]

### CODE REVIEW

**Comment by couet** [comment]
is this PR still valid ? https://root.cern/doc/master/df103__NanoAODHiggsAnalysis_8py.html seems ok ...

**Comment by hageboeck** [comment]
I think it is still valid:
image
image

and so on. Also, if I remember correctly, the notebook generation ran in 1h instead of 6h or so, so maybe that's a nice improvement for the nightly documentation jobs.

There were a few notebook failures even with those fixes IIRC, but it should be less than right now.

Should I rebase the branch for you?

@Axel-Naumann 
@couet

**Comment by couet** [comment]
yes please rebase. 
I will approve then you can merge

**Review by couet — APPROVED** [review]

File: documentation/doxygen/filter.cxx
Comment (couet) [inline documentation/doxygen/filter.cxx:199]: I do not understand this +0 you added

File: documentation/doxygen/filter.cxx
Comment (couet) [inline documentation/doxygen/filter.cxx:352]: I do not understand this +0 you added

File: documentation/doxygen/filter.cxx
Comment (couet) [inline documentation/doxygen/filter.cxx:368]: I do not understand this +0 you added

File: documentation/doxygen/filter.cxx
Comment (couet) [inline documentation/doxygen/filter.cxx:371]: I do not understand this +0 you added

File: documentation/doxygen/libs.C
Comment (couet) [inline documentation/doxygen/libs.C:17]: Why removing that code ?

File: documentation/doxygen/converttonotebook.py
Comment (jalopezg-r00t) [inline documentation/doxygen/converttonotebook.py:488]: Do we need to match `tailRe` here?  Shouldn't `findBracedBlock()` guarantee that `curlyEnd` always matches the closing brace at `curlyBegin`?

**Review by jalopezg-r00t — COMMENTED** [review]
Many thanks for the improvements, @hageboeck!  Just some comments here and there.

Can we rebase on top of `master` and check that it is still working?  After that, can we merge that, @couet?

File: documentation/doxygen/filter.cxx
Comment (hageboeck) [reply documentation/doxygen/filter.cxx:201]: Exactly. This makes a big difference for tutorials like RDataframe.

File: documentation/doxygen/filter.cxx
Comment (hageboeck) [reply documentation/doxygen/filter.cxx:307]: That's correct. There was a use for it maybe at some point (or it's used also somewhere else and here to show to possible future maintainers that both lists exist -- I don't remember). I can remove it if you want.

File: documentation/doxygen/converttonotebook.py
Comment (hageboeck) [reply documentation/doxygen/converttonotebook.py:422]: Yes, please do so! I had enough other problems to solve here already, so if you would want to look into that, it would be of great help!

File: documentation/doxygen/converttonotebook.py
Comment (hageboeck) [reply documentation/doxygen/converttonotebook.py:488]: I don't remember.

File: documentation/doxygen/converttonotebook.py
Comment (hageboeck) [reply documentation/doxygen/converttonotebook.py:829]: There were a lot of things that went wrong intermittently. For one, an RDataFrame notebook doesn't manage to connect to its input file.

**Comment by hageboeck** [comment]
@jalopezg-r00t I tried my best to rebase the old work. Hope this works now. I strongly recommend someone checks out and runs this on the machine that makes the notebooks and the documentation.

File: documentation/doxygen/Makefile
Comment (hageboeck) [reply documentation/doxygen/Makefile:51]: I removed this now. I anyway had to do a new rebase, since I overlooked a merge conflict marker on the first try.

File: documentation/doxygen/filter.cxx
Comment (hageboeck) [reply documentation/doxygen/filter.cxx]: Hopefully better now. The file should be reformatted anyway.

File: documentation/doxygen/filter.cxx
Comment (jalopezg-r00t) [reply documentation/doxygen/filter.cxx:307]: No worries; I have seen that the file being created is used somewhere else.

**Comment by jalopezg-r00t** [comment]
@hageboeck I tried it locally and seems to be working as intended - I saw some errors though, but I think those are related to my environment.
@couet I think it is essentially good to go :slightly_smiling_face:.  Could you pay attention after merging this that things are working as expected in the machine where the documentation is being built?

**Review by jalopezg-r00t — APPROVED** [review]
LGTM!

**Pull request merged**
